test/TestProjects/CadlFirstTest/Generated/HelloWorldClient.cs

```diff
@@ -78,16 +83,20 @@ public virtual async Task<Response<Thing>> TopActionValueAsync(string action, Ca
         }
 
         /// <param name="action"> The String to use. </param>
+        /// <param name="exEnum"> The ExtensibleEnum to use. </param>
+        /// <param name="enum"> The SimpleEnum to use. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="action"/> is null. </exception>
-        public virtual Response<Thing> TopActionValue(string action, CancellationToken cancellationToken = default)
+        public virtual Response<Thing> TopActionValue(string action, ExtensibleEnum exEnum, SimpleEnum? @enum = null, CancellationToken cancellationToken = default)
         {
+            Argument.AssertNotNull(action, nameof(action));
```

Comment:
@annelo-msft We must validate all the parameters that are validated in protocol methods. Otherwise, in case of incorrect path parameter value, we will open the scope and serialize body parameter into `RequestContent` before path parameter is validated.

Reply:
Parameters are validated outside diagnostic scopes being opened in protocol methods, but yes, in the case we need to open the scope in the convenience method you are right.  To make it a simple rule to follow let's go ahead and validate parameters in convenience methods as well, following the same rules for parameters that are validated in protocol methods.  I'll update relevant docs.  Thanks!